src/cr/cube/dimension.py

```diff
@@ -401,6 +305,208 @@ def _update_with_subtotals(self, labels_with_cat_ids):
 
         return labels_with_cat_ids
 
+    @lazyproperty
+    def _valid_elements(self):
+        """_Elements object providing access to non-missing elements.
+
+        Any categories or subvariables representing missing data are excluded
+        from the collection; this sequence represents a subset of that
+        provided by `._all_elements`.
+        """
+        return self._all_elements.valid_elements
```

Comment:
From this expression it seems that having "elements" repeated two times is redundant. Maybe something like
```
self._elements.all
```
vs
```
self._elements.valid
```
would be more concise and/or readable.
But this is just a comment, not sure it's good/easy to change it like that.

Reply:
I think this is the right naming. This location is the only place we expect this expression to arise and I'm strongly inclined to keep things explicit. If we neglect to distinguish all from valid here it makes readability poor because you have to remember (or more likely go look) to determine what that simplified name means.

End users will have no direct access to the _AllElements object so they will not need to use the `_AllElements.valid_elements` property.